src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/maintenance.html.twig

```diff
@@ -25,41 +25,22 @@
 {% extends '@PrestaShop/Admin/layout.html.twig' %}
 {% trans_default_domain "Admin.Shopparameters.Feature" %}
 {% import '@PrestaShop/Admin/macros.html.twig' as ps %}
-
+{% form_theme generalForm 'PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit.html.twig' %}
 
 {% block content %}
   {{ form_start(generalForm, {'attr' : {'class': 'form', 'id': 'form-maintenance'} }) }}
     <div class="row justify-content-center">
-      {% block maintenance_form_general %}
         <div class="col-xl-10">
           <div class="card">
             <h3 class="card-header">
               <i class="material-icons">business_center</i> {{ 'General'|trans({}, 'Admin.Global') }}
             </h3>
             <div class="card-block row">
               <div class="card-text">
-                <div class="form-group row">
-                  {{ ps.label_with_help(('Enable Shop'|trans), ('Activate or deactivate your shop (It is a good idea to deactivate your shop while you perform maintenance. Please note that the webservice will not be disabled).'|trans({}, 'Admin.Shopparameters.Help'))) }}
-                  <div class="col-sm">
-                    {{ form_errors(generalForm.enable_shop) }}
-                    {{ form_widget(generalForm.enable_shop) }}
-                  </div>
-                </div>
-                <div class="form-group row">
-                  {{ ps.label_with_help(('Maintenance IP'|trans), ('IP addresses allowed to access the front office even if the shop is disabled. Please use a comma to separate them (e.g. 42.24.4.2,127.0.0.1,99.98.97.96).'|trans({}, 'Admin.Shopparameters.Help'))) }}
-                  <div class="col-sm">
-                    {{ form_errors(generalForm.maintenance_ip) }}
-                    {{ form_widget(generalForm.maintenance_ip, { 'currentIp': currentIp}) }}
-                  </div>
-                </div>
-                <div class="form-group row">
-                  {{ ps.label_with_help(('Custom maintenance text'|trans), ('Custom text displayed on maintenance page while shop is deactivated.'|trans({}, 'Admin.Shopparameters.Help'))) }}
-                  <div class="col-sm">
-                    {{ form_errors(generalForm.maintenance_text) }}
-                    {{ form_widget(generalForm.maintenance_text) }}
-                  </div>
-                </div>
-                {{ form_rest(generalForm) }}
+                  {{ form_row(generalForm.enable_shop) }}
```

Comment:
Since you use the latest form theme this should be more simple, you should be able to use:
```
form_widget(generalForm)
```

And I guess `currentIp` should be passed to the form via an option (and possibly the `buildView` function) instead of via the template

Reply:
Yes 😄  and if using `form_widget()` creates some issues, you can ask @JevgenijVisockij about it if needed, he's the best